### PR TITLE
feat(Command): Added a few methods to the command class to check inputs.

### PR DESCRIPTION
**Pull request by built-tools**

### CODE REVIEW

File: honeybee_radiance_command/_command.py [hunk 1]
```diff
@@ -185,3 +185,95 @@ def after_run(self):
 
     def __repr__(self):
         return self.to_radiance()
+
+    @staticmethod
+    def path_checker(file_path=None, extn_list=[], file_descr=''):
```
Comment (mostaphaRoudsari) [inline honeybee_radiance_command/_command.py:190]: You should never use a mutable default value for an argument. See here: https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

Just use None and replace it with [] inside the code.

Comment (mostaphaRoudsari) [reply honeybee_radiance_command/_command.py:190]: Also in what case would file_path be None?

```suggestion
    def path_checker(file_path, extn_list=None, file_descr=None):
```

File: honeybee_radiance_command/_command.py [hunk 2]
```diff
@@ -185,3 +185,95 @@ def after_run(self):
...
+            extn_list: List of extensions, including the leading dot e.g. ['.hdr','.pic']
+                which if provided will be checked for before the path is set. If there
+                is no match, a value error will be raised.
+            file_descr: A phrase describing the file (e.g. 'Radiance HDR') that can be
+                used to compose the error message in case the extension check fails.
+        """
+
+        if file_path is None:
+            return None
+        else:
```
Comment (mostaphaRoudsari) [inline honeybee_radiance_command/_command.py:206]: As I said above you don't need this check.

File: honeybee_radiance_command/_command.py [hunk 3]
```diff
@@ -185,3 +185,95 @@ def after_run(self):
...
+            file_name,file_extn=os.path.splitext(file_path)
```
Comment (mostaphaRoudsari) [inline honeybee_radiance_command/_command.py:207]: missing space - you can use autopep8 of flake8 to catch these issues automatically.

```suggestion
            file_name, file_extn=os.path.splitext(file_path)
```

File: honeybee_radiance_command/_command.py [hunk 4]
```diff
@@ -185,3 +185,95 @@ def after_run(self):
...
+            if extn_list and file_extn.lower() not in extn_list:
+                file_descr='for %s files'%file_descr if file_descr else ''
```
Comment (mostaphaRoudsari) [inline honeybee_radiance_command/_command.py:209]:
```suggestion
                file_descr = 'for %s files' % file_descr if file_descr else ''
```

File: honeybee_radiance_command/_command.py [hunk 5]
```diff
@@ -185,3 +185,95 @@ def after_run(self):
...
+                err_msg='The provided input (%s) is invalid %s as only a file with the ' \
+                        'following extensions is allowed: %s'%(file_path,file_descr,
```
Comment (mostaphaRoudsari) [inline honeybee_radiance_command/_command.py:211]:
```suggestion
                        'following extensions is allowed: %s' % (file_path, file_descr,
```

File: honeybee_radiance_command/_command.py [hunk 6]
```diff
@@ -185,3 +185,95 @@ def after_run(self):
...
+                                                               ','.join(extn_list))
+                raise ValueError(err_msg)
+            return typing.normpath(file_path)
+
+    @staticmethod
+    def path_checker_multiple(file_paths=[], extn_list=[], file_descr='',
```
Comment (mostaphaRoudsari) [inline honeybee_radiance_command/_command.py:217]: Same comments here. I'm not going to comment on all the styling issues.

File: tests/command_test.py [hunk 7]
```diff
@@ -30,3 +33,57 @@ class Rtrace(Command):
 
     rtrace = Rtrace()
     assert rtrace.command == 'rtrace'
+
+def test_path_checker():
+    test_path1=r'test\some directory\some_file.rad'
+    test_path2=r'test\some directory\some_file.bmp'
```
Comment (mostaphaRoudsari) [inline tests/command_test.py:39]: In all our tests we use the UNIX format.

```suggestion
    test_path1 = 'test/some directory/some_file.rad'
    test_path2 = 'test/some directory/some_file.bmp'
```

**Review by mostaphaRoudsari — CHANGES_REQUESTED** [review]
Hi @built-tools,

There are some styling issues in the code itself but before getting to those I wanted to understand where are you thinking to use these checks? We already have similar checks in honeybee but not for files: https://github.com/ladybug-tools/honeybee-core/blob/master/honeybee/typing.py

**Comment by built-tools** [comment]
Hi @mostaphaRoudsari ,

1. I was hoping to get a discussion started so I did not clean up the script for stylistic/PEP-8 type issues (sorry!). 
2. I think the [typing module in the same repository ](https://github.com/ladybug-tools/honeybee-radiance-command/blob/master/honeybee_radiance_command/_typing.py) would be a better home for these checks. (We can place them in the honeybee-core too, although I don't know if they will be useful outside of Radiance).
3. As to why these might be useful. Mostly for much code-reuse and better maintenance. Specifically: 
 (a) Literally every case of setting the CommandOptions is essentially a copy paste of the same logic. Like [here](https://github.com/ladybug-tools/honeybee-radiance-command/blob/master/honeybee_radiance_command/oconv.py#L43-L51), [here](https://github.com/ladybug-tools/honeybee-radiance-command/blob/master/honeybee_radiance_command/gendaymtx.py#L45-L53), [here](https://github.com/ladybug-tools/honeybee-radiance-command/blob/master/honeybee_radiance_command/falsecolor.py#L40-L48), [here](https://github.com/ladybug-tools/honeybee-radiance-command/blob/master/honeybee_radiance_command/evalglare.py#L53-L61), etc. We are just checking if an instance of 'something' and then instantiating a new instance of that 'something' in case the input is None. Which is what I am doing with the [instance_checker method.](https://github.com/ladybug-tools/honeybee-radiance-command/blob/e978ff37a98c2a7b715e3fcab54b93edf0d29922/honeybee_radiance_command/_command.py#L252-L278).  I think we'd be able to use this methods for other types of instance checking beyond CommandOptions as well.

 (b) The reason for adding file checkers are similar as well. Way to much duplication of logic in existing code and future code. 

I will delete this branch and send a fresh PR for the typing module.

File: honeybee_radiance_command/_command.py (hunk 1 above)
Comment (built-tools) [reply honeybee_radiance_command/_command.py:190]: @mostaphaRoudsari Actually the file path is None as the default argument for several methods that we have in classes. Examples: [1](https://github.com/ladybug-tools/honeybee-radiance-command/blob/master/honeybee_radiance_command/gendaymtx.py#L60-L65). But I agree that it need not be set as default None here.

**Comment by built-tools** [comment]
@mostaphaRoudsari I have incorporated the changes as discussed yesterday in the typing module instead and sent a PR for that: https://github.com/ladybug-tools/honeybee-radiance-command/pull/209. I am going to close this PR now.

File: honeybee_radiance_command/_command.py (hunk 1 above)
Comment (mostaphaRoudsari) [reply honeybee_radiance_command/_command.py:190]: Hi @built-tools! I see why you need to handle the cases for None. Makes sense. I would still don't put None here as the default value.

**Pull request closed without merging**